File: app/src/main/java/com/example/changfeng/taptapword/net/ApiClient.java
package com.example.changfeng.taptapword.net;


import com.example.changfeng.taptapword.Config;
import com.example.changfeng.taptapword.util.MD5Utils;
import com.example.changfeng.taptapword.util.Utils;
import com.example.changfeng.taptapword.net.result.BaiduResult;
import com.example.changfeng.taptapword.net.result.YoudaoResult;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Retrofit;
import retrofit2.converter.moshi.MoshiConverterFactory;

/**
 * Created by changfeng on 2016/3/17.
 */
public class ApiClient {
    private static final String TAG = "ApiClient";
    public static final String BAIDU_BASE_URL = "http://api.fanyi.baidu.com";
    public static final String YOUDAO_BASE_URL = "http://fanyi.youdao.com";

    Retrofit baiduRetrofit = new Retrofit.Builder()
            .baseUrl(BAIDU_BASE_URL)
            .addConverterFactory(MoshiConverterFactory.create())
            .build();

    Retrofit youdaoRetrofit = new Retrofit.Builder()
            .baseUrl(YOUDAO_BASE_URL)
            .addConverterFactory(MoshiConverterFactory.create())
            .build();

    BaiduFanyiService baiduFanyiService = baiduRetrofit.create(BaiduFanyiService.class);

    YoudaoFanyiService youdaoFanyiService = youdaoRetrofit.create(YoudaoFanyiService.class);

    public void getBaiduResult(String query, Callback<BaiduResult> callback) {
        query = query.replaceAll(" ", "%20");
        try {
            String salt = Utils.getToken();
            String sign = MD5Utils.getMd5(Config.baiduTranslateAppId + query + salt + Config.baiduTranslateAppKey);
            Call<BaiduResult> call = baiduFanyiService.getResult(query, "en", "zh", Config.baiduTranslateAppId, salt, sign);
            call.enqueue(callback);
        } catch (Exception e) {

        }
    }

    public void getYoudaoResult(String query, Callback<YoudaoResult> callback) {
        query = query.replaceAll(" ", "%20");
        try {
            Call<YoudaoResult> call = youdaoFanyiService.getResult(Config.youdaoDictKeyfrom, Config.youdaoDictApiKey, "data", "json", "1.1", query);
            call.enqueue(callback);
        } catch (Exception e) {

        }
    }
}

File: app/src/main/java/com/example/changfeng/taptapword/net/ApiClient.kt
package com.example.changfeng.taptapword.net


import android.util.Log
import com.example.changfeng.taptapword.Config
import com.example.changfeng.taptapword.net.result.BaiduResult
import com.example.changfeng.taptapword.net.result.YoudaoResult
import com.example.changfeng.taptapword.util.MD5Utils
import com.example.changfeng.taptapword.util.Utils
import retrofit2.Callback
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by changfeng on 2016/3/17.
 */
class ApiClient {

    private var baiduRetrofit = Retrofit
            .Builder()
            .baseUrl(BAIDU_BASE_URL).addConverterFactory(GsonConverterFactory.create()).build()

    private var youdaoRetrofit = Retrofit
            .Builder()
            .baseUrl(YOUDAO_BASE_URL)
            .addConverterFactory(GsonConverterFactory.create()).build()

    private var baiduFanyiService = baiduRetrofit.create(BaiduFanyiService::class.java)

    private var youdaoFanyiService = youdaoRetrofit.create(YoudaoFanyiService::class.java)

    fun getBaiduResult(query: String, callback: Callback<BaiduResult>) {
        try {
            val salt = Utils.getToken()
            val sign = MD5Utils.getMd5(Config.baiduTranslateAppId + query + salt + Config.baiduTranslateAppKey)
            val call = baiduFanyiService.getResult(query, "en", "zh", Config.baiduTranslateAppId, salt, sign)
//            Log.i(TAG, "getBaiduResult() " + call.request().url())
            call.enqueue(callback)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    fun getYoudaoResult(query: String, callback: Callback<YoudaoResult>) {
        try {
            val call = youdaoFanyiService.getResult(Config.youdaoDictKeyfrom, Config.youdaoDictApiKey, "data", "json", "1.1", query)
            Log.i(TAG, "getYoudaoResult() " + call.request().url())
            call.enqueue(callback)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    companion object {
        private const val TAG = "ApiClient"
        const val BAIDU_BASE_URL = "http://api.fanyi.baidu.com"
        const val YOUDAO_BASE_URL = "http://fanyi.youdao.com"
    }
}

